samples/snippets/src/main/java/pubsub/SubscribeWithAvroSchemaExample.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pubsub;
+
+// [START pubsub_subscribe_avro_records]
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.Encoding;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.TopicName;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.specific.SpecificData;
+import utilities.State;
+
+public class SubscribeWithAvroSchemaExample {
+  public static void main(String... args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    // Use a topic that has a schema attached.
+    String topicId = "your-topic-id";
+    // Use a subscription attached to your topic.
+    String subscriptionId = "your-subscription-id";
+
+    subscribeWithAvroSchemaExample(projectId, topicId, subscriptionId);
+  }
+
+  public static void subscribeWithAvroSchemaExample(String projectId, String topicId, String subscriptionId)
+      throws IOException {
+
+    TopicName topicName = TopicName.of(projectId, topicId);
+    ProjectSubscriptionName subscriptionName = ProjectSubscriptionName.of(projectId, subscriptionId);
+
+    // Get the topic encoding type.
+    Encoding encoding = null;
+    try (TopicAdminClient topicAdminClient = TopicAdminClient.create()) {
```

Comment:
It would be an anti-pattern for the subscriber to get properties from the topic. The encoding and schema name are passed as attributes in the message. googclient_schemaencoding and googclient_schemaname. This means the subscriber shouldn't need to fetch the info from the topic.

Reply:
I wasn't sure earlier if googclient_schemaencoding and googclient_schemaname are part of the emulator or the real service. Looks like they are. Updated.